src/mrseq/scripts/t1rho_se_single_line.py

```diff
@@ -0,0 +1,393 @@
+            seq.add_block(gz90_reph)
+
+            # add gradients and refocusing pulse
+            seq.add_block(pp.make_delay(delay_gz90_reph_and_gz_spoil))
+            seq.add_block(gz_spoil)
+            seq.add_block(rf180, gz180)
+            seq.add_block(gz_spoil)
+            seq.add_block(pp.make_delay(delay_gz_spoil_and_gx_pre))
+
+            # calculate phase encoding gradient for current phase encoding step
+            gy_pre = pp.make_trapezoid(channel='y', area=phase_areas[pe_idx], duration=gx_pre_duration, system=system)
+
+            # add pre-winder gradients and labels
+            seq.add_block(gx_pre, gy_pre, pe_label, contrast_label)
+
+            # add readout gradient and ADC
+            seq.add_block(gx, adc)
+
+            # add x and y re-winder and spoiler gradient in z-direction
+            gy_pre.amplitude = -gy_pre.amplitude
+            seq.add_block(gx_post, gy_pre, gz_spoil)
+
+            # calculate TR delay
+            duration_tr_block = sum(seq.block_durations.values()) - _start_time_tr_block
+            tr_delay = round_to_raster(tr - duration_tr_block, system.block_duration_raster)
+
+            # save time for sequence plot
+            if tsl_idx == 0 and pe_idx == 0:
+                time_to_first_tr_block = duration_tr_block
+
+            if tr_delay < 0:
+                raise ValueError('Desired TR too short for given sequence parameters.')
+
+            seq.add_block(pp.make_delay(tr_delay))
+
+    return seq, time_to_first_tr_block, min_te
+
+
+def main(
+    system: pp.Opts | None = None,
+    spin_lock_times: np.ndarray | None = None,
+    te: float | None = None,
+    tr: float = 8,
+    fov_xy: float = 128e-3,
+    n_readout: int = 128,
+    n_phase_encoding: int = 128,
+    slice_thickness: float = 8e-3,
+    show_plots: bool = True,
+    test_report: bool = True,
+    timing_check: bool = True,
+) -> pp.Sequence:
+    """Generate a SE-based sequence for T1rho mapping with one preparation pulse before every readout.
+
+    Parameters
+    ----------
+    system
+        PyPulseq system limits object.
+    spin_lock_times
+        Array of inversion times (in seconds).
+        Default values [0.025, 0.050, 0.1] s are used if set to None.
+    te
+        Desired echo time (TE) (in seconds). Minimum echo time is used if set to None.
+    tr
+        Desired repetition time (TR) (in seconds).
+    fov_xy
+        Field of view in x and y direction (in meters).
+    n_readout
+        Number of frequency encoding steps.
+    n_phase_encoding
+        Number of phase encoding steps.
+    slice_thickness
+        Slice thickness of the 2D slice (in meters).
+    show_plots
+        Toggles sequence plot.
+    test_report
+        Toggles advanced test report.
+    timing_check
+        Toggles timing check of the sequence.
+    """
+    if system is None:
+        system = sys_defaults
+
+    if spin_lock_times is None:
+        spin_lock_times = np.array([0.025, 0.050, 0.1])
```

Comment:
100 ms should be the upper limit for the duration of a single RF pulse at Siemens. However, last time I tested the spin-lock sequence on the Cima.X I had to reduce the max TSL to 99 ms. Let's try it with the 100 ms in this sequence and reduce it, if we realize it's not working. M